text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -267,15 +267,19 @@ def fit(self,
         else:
             label_columns = list(self._label)
         # Get the training and tuning data as pandas dataframe
-        if not isinstance(train_data, pd.DataFrame):
+        if isinstance(train_data, str):
```

Comment:
@sxjscience did you use the old implementation in a more advanced way or is this change fine?

Reply:
This is fine.